texar/modules/encoders/bert_encoders.py

```diff
@@ -0,0 +1,346 @@
+    :class:`~texar.modules.encoders.TransformerEncoder` and a dense pooler.
+
+    This module supports the architecture first proposed
+    in `(Devlin et al.)` BERT.
+
+    Args:
+        pretrained_model_name (optional): a str with the name
+            of a pre-trained model to load selected in the list of:
+            `bert-base-uncased`, `bert-large-uncased`, `bert-base-cased`,
+            `bert-large-cased`, `bert-base-multilingual-uncased`,
+            `bert-base-multilingual-cased`, `bert-base-chinese`.
+            If `None`, will use the model name in :attr:`hparams`.
+        cache_dir (optional): the path to a folder in which the
+            pre-trained models will be cached. If `None` (default),
+            a default directory will be used.
+        hparams (dict or HParams, optional): Hyperparameters. Missing
+            hyperparameter will be set to default values. See
+            :meth:`default_hparams` for the hyperparameter sturcture
+            and default values.
+
+    .. document private functions
+    .. automethod:: _build
+    """
+    def __init__(self,
+                 pretrained_model_name=None,
+                 cache_dir=None,
+                 hparams=None):
+        EncoderBase.__init__(self, hparams)
+        BertBase.__init__(self, pretrained_model_name,
+                          cache_dir, hparams)  # put these things to BertBase
+        if self.pretrained_model:
+            self._hparams = HParams(self.pretrained_model_hparams,
+                                    self._hparams.todict())
+
+        with tf.variable_scope(self.variable_scope):
+            if self._hparams.initializer:
+                tf.get_variable_scope().set_initializer(
+                    layers.get_initializer(self._hparams.initializer))
+
+            # Word embedding
+            self.word_embedder = WordEmbedder(
+                vocab_size=self._hparams.vocab_size,
+                hparams=self._hparams.embed)
+
+            # Segment embedding for each type of tokens
+            self.segment_embedder = WordEmbedder(
+                vocab_size=self._hparams.type_vocab_size,
+                hparams=self._hparams.segment_embed)
+
+            # Position embedding
+            self.position_embedder = PositionEmbedder(
+                position_size=self._hparams.position_size,
+                hparams=self._hparams.position_embed)
+
+            # The BERT encoder (a TransformerEncoder)
+            self.encoder = TransformerEncoder(hparams=self._hparams.encoder)
+
+            with tf.variable_scope("pooler"):
+                kwargs_i = {"units": self._hparams.hidden_size,
+                            "activation": tf.tanh}
+                layer_hparams = {"type": "Dense", "kwargs": kwargs_i}
+                self.pooler = layers.get_layer(hparams=layer_hparams)
+
+    @staticmethod
+    def default_hparams():
+        """Returns a dictionary of hyperparameters with default values.
+
```

Comment:
Add the following docstring:

```
* The encoder arch is determined by the constructor argument :attr:`pretrained_model_name` if it's specified. In this case, hparams are ignored.
* Otherwise, the encoder arch is determined by `hparams['pretrained_model_name'] if it's specified. all other configs in hparams are ignored.`
* If the above two are `None`, the encoder arch is defined by the configs in hparams and weights are randomly initialized.
```